File: ARRAY/check_if_arrayisSORTED.cpp
#include<bits/stdc++.h>
using namespace std;
//USING BRUTE FORCE APPROACH TC---O(N) SC---O(1)
bool sort(vector<int> &arr,int n){
int b=true;
    for(int i=1;i<n;i++){
        if(!(arr[i-1]<=arr[i]))
            b=false;
        }
    
    return b;
    }

int main(){
    int n;
    cin>>n;
    vector<int>arr(n);
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
   cout<< sort(arr,n);
}

File: Recursion/CheckifArrayisSorted.java
public class CheckifArrayisSorted {
    public static void main(String[] args) {
        int[] arr={1,2,3,4,5};
        System.out.println(        f(0,arr)
);
    }
    static boolean f(int i,int[] arr){
        int n=arr.length;
        if(i==n-1)return true;
        if(!(arr[i]<=arr[i+1])){
            return false;

        }
        return f(i+1, arr);


    }
}
